messageparser.go

```diff
@@ -613,6 +630,54 @@ func parseResponse(logMessage *LogMessage, key string, value string) error {
 	return nil
 }
 
+// parseEndpointProperty is a custom function to parse the EndpointProperty field into the EndpointProperty field of the LogMessage struct
+func parseEndpointProperty(logMessage *LogMessage, key string, value string) error {
+
+	endpointProperty := &LogMessage{
+		MessageDetails: MessageDetails{
+			UnexpectedFields: map[string]string{},
+		},
+	}
+	endpointProperty.NetworkDeviceGroups = DropDownMap{}
+
+	endpointPropertySlice := strings.Split(value, `\,`)
+	if len(endpointPropertySlice) == 0 {
+		return &TypeMismatch{
+			Original: endpointPropertySlice,
+			Type:     "EndpointProperty",
+		}
+	}
+
+	var keyValueEndpointPropertyFuncMap = parseFnMap{
+		"textEncodedORAddress": parseDisregard, // Contents differ and cannot use its parse func. Information is not valuable in 80002 event so ignore
+		"#015":                 parseDisregard,
+		"cisco-av-pair":        parseCiscoAVPair,
+		"device-type":          parseDisregard, // Different type than LogMessage's DeviceType. Since it's only populated for individually owned assets (data we can find in TAG project), we can ignore
+		"Response":             parseResponse,
+	}
+
+	for _, column := range endpointPropertySlice {
+		key, value := extractKeyValue(column)
+
+		parseFunc := keyValueEndpointPropertyFuncMap.retrieveParseFn(endpointProperty, key)
+		err := parseFunc(endpointProperty, formatKey(key), value)
+		if err != nil {
+			if assignmentFailure, ok := err.(*AssignmentFailure); ok && strings.Contains(assignmentFailure.Message, "invalid-value") {
+				addUnexpectedKeyValue(endpointProperty, key, value)
+				continue
+			}
+			return &ParseError{
+				OrigErr: err,
+				Message: "parse-field-error",
+				Reason:  fmt.Sprintf("could not parse key - value: %s - %v", key, value),
+			}
+		}
+
+		logMessage.EndpointProperty = endpointProperty
```

Comment:
Nit, this statement is being set on each iteration of the for loop. You should move it to be outside of it

Reply:
oops! Agreed